app/ante/integration_test.go

```diff
@@ -0,0 +1,82 @@
+package ante_test
+
+import (
+	"log"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/evmos/evmos/v11/crypto/ethsecp256k1"
+	"github.com/evmos/evmos/v11/testutil"
+	testutiltx "github.com/evmos/evmos/v11/testutil/tx"
+	"github.com/evmos/evmos/v11/utils"
+)
+
+var _ = Describe("when sending a Cosmos transaction", func() {
+	var (
+		addr sdk.AccAddress
+		priv *ethsecp256k1.PrivKey
+		msg  sdk.Msg
+	)
+
+	BeforeEach(func() {
+		s.SetupTest()
+
+		addr, priv = testutiltx.NewAccAddressAndKey()
+		msg = &banktypes.MsgSend{
+			FromAddress: addr.String(),
+			ToAddress:   "evmos1dx67l23hz9l0k9hcher8xz04uj7wf3yu26l2yn",
+			Amount:      sdk.Coins{sdk.Coin{Amount: sdkmath.NewInt(1e14), Denom: utils.BaseDenom}},
+		}
+	})
+
+	Context("and the sender account has enough balance to pay for the transaction cost", Ordered, func() {
+		BeforeAll(func() {
+			s.ctx, _ = testutil.PrepareAccountsForDelegationRewards(
+				s.T(), s.ctx, s.app, addr, sdk.NewInt(1e16), sdk.NewInt(1e16),
+			)
+			log.Printf("\nAccount balance: %s", s.app.BankKeeper.GetBalance(s.ctx, addr, utils.BaseDenom).Amount.String())
+		})
+
+		It("should succeed", func() {
+			// FIXME: this test fails because apparently the balance is not sufficient upon calling DeliverTx, but as
+			// it can be seen here, it's not zero?
+			log.Printf("\nAccount balance: %s", s.app.BankKeeper.GetBalance(s.ctx, addr, utils.BaseDenom).Amount.String())
+			_, err := testutil.DeliverTx(s.ctx, s.app, priv, nil, msg)
+			Expect(err).To(BeNil())
```

Comment:
really not sure why this test fails with apparently no balance, because the logger clearly says there is a balance there..